frontend/app/routes/protected/multi-channel/edit-application/validation.server.ts

```diff
@@ -0,0 +1,114 @@
+import * as v from 'valibot';
+
+import { getSinCaseService } from '~/.server/domain/multi-channel/case-api-service';
+import type { SinCaseDto } from '~/.server/domain/multi-channel/sin-case-models';
+import { i18nRedirect } from '~/.server/utils/route-utils';
+import { HttpStatusCodes } from '~/errors/http-status-codes';
+import { primaryDocument } from '~/routes/protected/sin-application/validation.server';
+import { toISODateString } from '~/utils/date-utils';
+
+export async function fetchSinCase(session: AppSession, caseId: string) {
+  if (session.editingCase && session.editingCase.caseId === caseId) {
+    return session.editingCase;
+  }
+  const sinCase = await getSinCaseService().getSinCaseById(caseId);
```

Comment:
This file's responsibility is validation. In my opinion, it shouldn't be fetching data unless it's some lookup code that is required for validation.